packages/react-utilities/src/compose/getSlots.ts

```diff
@@ -86,7 +82,7 @@ export function getSlots<SlotProps extends SlotPropsRecord = {}>(state: Componen
   const [root, rootProps] = getRootSlot(state);
 
   const typedSlotProps = slotProps as {
-    [Key in keyof SlotProps]: UnionToIntersection<SlotProps[Key]>;
+    [Key in keyof SlotProps]-?: UnionToIntersection<NonNullable<SlotProps[Key]>>;
```

Comment:
curious: what is `-?` ?

Reply:
Making all the keys required https://www.typescriptlang.org/docs/handbook/2/mapped-types.html#mapping-modifiers